### PR TITLE
Task 7 post new comment route

### PR DESCRIPTION
**Pull request by VadimAlexeev0**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -132,6 +132,32 @@ describe("GET: /api/reviews/:review_id/comments", ()=>{
...
+
+        return request(app)
+            .post("/api/reviews/1/comments")
+            .send(requestBody)
+            .expect(201)
+            .then(({body})=>{
+                const { newComment } = body;
+                
+                expect(newComment).toMatchObject({
+                    author: expect.any(String),
```
Comment (roisinoreilly) [inline __tests__/app.test.js:150]: can you check here that the new comment posted has the author and body that you've given it

File: __tests__/app.test.js [hunk 2]
```diff
@@ -171,4 +197,25 @@ describe("Error Handling", ()=>{
                 })
         })
     })
+
+    describe("POST: /api/reviews/:review_id/comments", ()=>{
```
Comment (roisinoreilly) [inline __tests__/app.test.js:201]: this should probably be in the same describe block as on line 135 if you're testing the same endpoint

File: __tests__/app.test.js [hunk 3]
```diff
@@ -171,4 +197,25 @@ describe("Error Handling", ()=>{
...
+        test("400: When given non numerical review_id", ()=>{
+            return request(app)
+                .post("/api/reviews/NaN/comments")
+                .send({username: "test", body: "test"})
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe("400 Invalid input")
+                })
+        })
+        test("400: When missing key in body", ()=>{
```
Comment (roisinoreilly) [inline __tests__/app.test.js:211]: nice edge case

File: __tests__/app.test.js [hunk 4]
```diff
@@ -171,4 +197,25 @@ describe("Error Handling", ()=>{
                 })
         })
     })
+
+    describe("POST: /api/reviews/:review_id/comments", ()=>{
+        test("400: When given non numerical review_id", ()=>{
+            return request(app)
+                .post("/api/reviews/NaN/comments")
```
Comment (roisinoreilly) [inline __tests__/app.test.js:204]: you're actually testing two different things in this assertion, you're testing an invalid review id AND a username that isn't in the database (which is actually a 404). Extract these into two different tests and make sure you're getting back the status codes you expect

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
a great start! There's a couple more tests I'd like you to add, separate the invalid id test out from the invalid username test, you might need to make a separate call to the database to check if a username is valid. Can you also test for a valid but non existent id eg. /reviews/9999/comments. When you've added those send us another PR :)

**Comment by VadimAlexeev0** [comment]
Commit might be a bit confusing since I moved the error tests to be inside the same describe block
